client/src/utils/fetchUserData.js

```diff
@@ -40,3 +42,20 @@ export async function fetchProfileData(user) {
     console.error(err);
   }
 }
+
+export async function fetchRegister(body) {
```

Comment:
I am a little confused by the logic of this code block, maybe some comments might help. I think also there might be too much in the one try block when there are multiple things happening to go wrong imo. Would appreciate your thoughts on this.

Reply:
yikes ok good to know